backend/src/main/java/ai/giskard/security/PermissionEvaluator.java

```diff
@@ -26,14 +28,18 @@ public boolean isCurrentUser(String login) {
     }
 
     /**
-     * Determine if a user can write a project, i.e. is admin or project's owner
+     * Determine if a user can write a project, i.e. is admin, is project's owner or is guest with any role
```

Comment:
This is not correct and grants too many permissions to guests

"canWriteProject" is a very global setting and impact way more than test suites. This change effectively gives admin rights over a project to any guest inside it.

The correct change in this case would be to have a specific check on test suite creation. In that place, we would make sure that the guest user has either the AI Tester or AI Creator role